tests/mezflipr.py

```diff
@@ -46,43 +38,63 @@ def setUp(self):
-        for val in [0., 0.12, 2.99]:  # Amplitude should be limited to 3A
-            self.ca.assert_setting_setpoint_sets_readback(val, readback_pv="{}:AMPLITUDE".format(flipper))
-
-    @parameterized.expand(["ANALYSER", "POLARISER"])
-    def test_GIVEN_compensation_is_set_THEN_compensation_can_be_read_back(self, flipper):
-        for val in [0., 0.12, 5000.5]:
-            self.ca.assert_setting_setpoint_sets_readback(val, readback_pv="{}:COMPENSATION".format(flipper))
-
-    @parameterized.expand(["ANALYSER", "POLARISER"])
-    def test_GIVEN_dt_is_set_THEN_dt_can_be_read_back(self, flipper):
-        for val in [0., -0.12, -5000.5]:  # DT only accepts negative values.
-            self.ca.assert_setting_setpoint_sets_readback(val, readback_pv="{}:DT".format(flipper))
-
-    @parameterized.expand(["ANALYSER", "POLARISER"])
-    def test_GIVEN_constant_is_set_THEN_constant_can_be_read_back(self, flipper):
-        for val in [0., 0.12, 5000.5]:
-            self.ca.assert_setting_setpoint_sets_readback(val, readback_pv="{}:CONSTANT".format(flipper))
-
-    @parameterized.expand(["ANALYSER", "POLARISER"])
-    def test_GIVEN_constant_is_set_THEN_constant_can_be_read_back(self, flipper):
-        for filename in [r"C:\a.txt", r"C:\b.txt"]:
-            self.ca.assert_setting_setpoint_sets_readback(filename, readback_pv="{}:FILENAME".format(flipper))
-
-    @parameterized.expand([
-        "Analyser Off Pol. Off",
-        "Analyser Off Pol. On",
-        "Analyser On Pol. Off",
-        "Analyser On Pol. On",
-    ])
-    def test_GIVEN_toggle_state_is_set_THEN_toggle_state_can_be_read_back(self, toggle_state):
-        self.ca.set_pv_value("TOGGLE:SP", toggle_state)
-        self.ca.assert_that_pv_is("TOGGLE", toggle_state)
-        self.ca.assert_that_pv_alarm_is("TOGGLE", self.ca.Alarms.NONE)
+    @parameterized.expand(parameterized_list(["On", "Off"]))
+    def test_GIVEN_power_is_set_THEN_can_be_read_back(self, _, state):
+        self.ca.assert_setting_setpoint_sets_readback(state, readback_pv="{}:POWER".format(flipper))
+
+    @parameterized.expand(parameterized_list([0., 0.12, 5000.5]))
+    def test_GIVEN_compensation_is_set_THEN_compensation_can_be_read_back(self, _, compensation):
+        self.ca.assert_setting_setpoint_sets_readback(compensation, readback_pv="{}:COMPENSATION".format(flipper))
+
+    @skip_if_recsim("State of device not simulated in recsim")
+    def test_WHEN_constant_current_mode_set_THEN_parameters_reflected_and_mode_is_static(self):
+        param = 25
+        self.ca.set_pv_value("{}:CURRENT:SP".format(flipper), param)
+
+        self.ca.assert_that_pv_is("{}:PARAMS".format(flipper), "{:.1f}".format(param))
+        self.ca.assert_that_pv_alarm_is("{}:PARAMS".format(flipper), self.ca.Alarms.NONE)
+
+        self.ca.assert_that_pv_is("{}:MODE".format(flipper), "static")
+        self.ca.assert_that_pv_alarm_is("{}:MODE".format(flipper), self.ca.Alarms.NONE)
```

Comment:
Does it make sense to pull these into a method as they're being done in many of the tests?